### PR TITLE
bookstack: 24.05.2 -> 24.05.4

### PR DESCRIPTION
**Pull request by HritwikSinghal**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by HritwikSinghal** [comment]
bookstack 24.05.4 is a security update as mentioned [here](https://www.bookstackapp.com/blog/bookstack-release-v24-05-4/)

**Comment by HritwikSinghal** [comment]
changelog since 24.05.2: https://github.com/BookStackApp/BookStack/compare/v24.05.2...v24.05.4
changelog for 24.05.4: https://github.com/BookStackApp/BookStack/releases/tag/v24.05.4
```
Security Release

    [Update Instructions](https://www.bookstackapp.com/docs/admin/updates)
    [Update details on blog](https://www.bookstackapp.com/blog/bookstack-release-v24-05-4/)

BookStack v24.05.4 has been released.

This is a security release to address issues found in LDAP group syncing, where in certain scenarios a user could be matched to extra roles incorrectly, and an issue with content visibility in "book-show" API responses which would not have permissions applied properly.

Upgrade is strongly advised for instances where LDAP authentication is used with group syncing, or where the REST API is used to fetch contents of books ("books-read" endpoint).

Thanks to Linus Nagel and their team at WorkSimple GmbH for reporting this API vulnerability.
Full List of Changes

    Updated API docs with consistent parameter types. (https://github.com/BookStackApp/BookStack/issues/5183)
    Updated default content iframe embed max-width to align with other content types. (https://github.com/BookStackApp/BookStack/issues/5130)
    Updated LDAP group sync to query via full DN.
    Updated translations with latest Crowdin changes. (https://github.com/BookStackApp/BookStack/pull/5118)
    Fixed books read API response not applying visibility control to chapter contents.
    Fixed API docs users response showing extra property. (https://github.com/BookStackApp/BookStack/issues/5178)
    Fixed database error thrown when using out dev docker setup. (https://github.com/BookStackApp/BookStack/issues/5124)
    Fixed RTL display issues with tasklist checkboxes. (https://github.com/BookStackApp/BookStack/issues/5134)

```

**Comment by HritwikSinghal** [comment]
last PR https://github.com/NixOS/nixpkgs/pull/322136/

**Comment by HritwikSinghal** [comment]
can we also backport this?

File: pkgs/servers/web-apps/bookstack/default.nix [hunk 1]
```diff
@@ -16,13 +16,13 @@ let
...
   pname = "bookstack";
-  version = "24.05.2";
+  version = "24.05.4";
 
   src = fetchFromGitHub {
     owner = "bookstackapp";
     repo = pname;
     rev = "v${version}";
-    sha256 = "13yln6b5j9plkcrp1mwfvgcv67gm2ayb2l2grhw2h4ncqnvmxhwv";
+    sha256 = "1aa4im2khxycv5i2ff23ss91p4kr7d2a4yhjm0k0n6av64zhz2x3";
```
Comment (SuperSandro2000) [inline pkgs/servers/web-apps/bookstack/default.nix:25]:
```suggestion
    hash = "sha256-o4sPPzFbGQsmqBJ6okQ7eZIbktZDOCdi2cx3OEWNRKk=";
```

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
> can we also backport this?

sure, just add the label for it if you think it should be

**Comment by HritwikSinghal** [comment]
> > can we also backport this?
> 
> sure, just add the label for it if you think it should be

Thanks for quickly reviewing and merging it. will add label for backport. What should i do after cherry-pick? should i open another PR?

**Comment by phanirithvij** [comment]
Since the bot failed, yes you should send a pr targeting release-24.05 branch. See for eg. https://github.com/NixOS/nixpkgs/pull/339210

**Comment by HritwikSinghal** [comment]
> Since the bot failed, yes you should send a pr targeting release-24.05 branch. See for eg. #336734
> 
> https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#manually-backporting-changes

https://github.com/NixOS/nixpkgs/pull/339492

can you merge this.

**Pull request merged**